### ISSUE TITLE
Make Error at CMakeLists.txt:42 (FIND_PACKAGE):

### ISSUE THREAD

**Issue by darjamfar** [google-research/football#306]
I'm trying install google football. I issued command on my Mac and get the following error(s)

% python3 -m pip install gfootball

-- Found OpenGL: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/System/Library/Frameworks/OpenGL.framework
    CMake Error at CMakeLists.txt:42 (FIND_PACKAGE):
      By not providing "FindSDL2.cmake" in CMAKE_MODULE_PATH this project has
      asked CMake to find a package configuration file provided by "SDL2", but
      CMake did not find one.
    
      Could not find a package configuration file provided by "SDL2" with any of
      the following names:
    
        SDL2Config.cmake
        sdl2-config.cmake
    
      Add the installation prefix of "SDL2" to CMAKE_PREFIX_PATH or set
      "SDL2_DIR" to a directory containing one of the above files.  If "SDL2"
      provides a separate development package or SDK, be sure it has been
      installed.

-- Configuring incomplete, errors occurred!
    .../CMakeFiles/CMakeOutput.log".
   .../CMakeFiles/CMakeError.log".

    gfootball/build_game_engine.sh: line 29: pushd: third_party/gfootball_engine: No such file or directory

How do I fix this please ? 

(Edit. I should say I downloaded SDL2 and putting the framework in /Library/Frameworks. But this has made no difference
I get the exact same errors if I do the following:
$ git clone https://github.com/google-research/football.git
$ cd football
$ python3 -m pip install )

**Comment by vi3itor** [comment]
Hi!

What CMake version do you have? And how did you install it?
The easiest way is to follow [the installation guide](https://github.com/google-research/football#macos) and use `brew` to install all the dependencies: `brew install cmake sdl2` and so on.

If you don't want to use `brew` or need a specific version of SDL2 you'll have to make sure that CMake knows where to find it. One of the ways is to put a `FindSDL2.cmake` file into `third_party/gfootball_engine/CMakeModules` directory. An example of such file can be found [here](https://github.com/aminosbh/sdl2-cmake-modules), but don't forget to edit it according to your setup.

In any case, I strongly advise you to consider using `brew` because you might face additional problems with `Boost.Python`.

**Comment by darjamfar** [comment]
Hi! Thank you for replying.

On the macOS instructions, it instructs to install homebrew. Then to install the required packages. Now, originally I copied only the first line (brew install git python3 cmake sdl2 sdl2_image sdl2_ttf sdl2_gfx boost boost-python3) - not realising the second line was additional to the first. From there I was performing the other instructions to get the error msg on my original post.

Now, when I tried the instructions again, realising my error, I was getting weird errors like:

error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree. A bit of googling led me to believe deleting homebrew was the best action.

So I decided to delete homebrew  directories (/usr/local/Homebrew /Caskroom /Cellar and start again !

I've installed homebrew from the link. I've copied the full command. I also performed command associated with symlinks from output text.

brew install git python3 cmake sdl2 sdl2_image sdl2_ttf sdl2_gfx boost boost-python3 python3 -m pip install --upgrade pip setuptools psutil wheel
python3 -m pip install gfootball. <!! This command gives me same error as original post>

I also tried 

git clone https://github.com/google-research/football.git 
cd football
python3 -m pip install .  < This gives me same error as original post>

cmake --version = 3.22.1
sdl2 --version = command not found

Is the problem brew is not installing sdl2 ?

HTH, Thanks.

**Comment by darjamfar** [comment]
Hi, Thanks for replying.

I originally did not enter the installation command correctly. The command ...

brew install git python3 cmake sdl2 sdl2_image sdl2_ttf sdl2_gfx boost boost-python3 
python3 -m pip install --upgrade pip setuptools psutil wheel

... I stopped after first line, not realising the second was additional. When I tried full command I was getting weird errors like:

error: Not a valid ref: refs/remotes/origin/master
 fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.

A bit of googling led me to believe that deleting homebrew was best course of action was to delete homebrew, which I did.

I started again in a different directory and now am getting a different error but the same flavour:

CMake Error at CMakeModules/FindPackageHandleStandardArgs.cmake:110 (MESSAGE):
      Could NOT find SDL2_image (missing: SDL2_IMAGE_LIBRARIES
      SDL2_IMAGE_INCLUDE_DIRS)
    Call Stack (most recent call first):
      CMakeModules/FindPackageHandleStandardArgs.cmake:301 (_FPHSA_FAILURE_MESSAGE)
      CMakeModules/FindSDL2_image.cmake:104 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      CMakeLists.txt:55 (FIND_PACKAGE)

cmake --version is 3.22.1
sdl2 --version = not found, but looking in /usr/local/Cellar/sdl2/ there is something there called  2.0.18

HTH.
